builder/image/slugbuilder/builder/build.sh

```diff
@@ -76,7 +76,15 @@ if [[ -n "$BUILDPACK_URL" ]]; then
 
     buildpack="$buildpack_root/custom"
     rm -fr "$buildpack"
-    git clone --quiet --depth=1 "$BUILDPACK_URL" "$buildpack"
+
+    url=${BUILDPACK_URL%#*}
+    branch=${BUILDPACK_URL#*#}
+    if [[ "$branch" == "$url" ]] || [[ -z "${branch}" ]]; then
+        branch="master"
+    fi
+
+    git clone --quiet --branch "$branch" --depth=1 "$url" "$buildpack"
```

Comment:
I think you could re-use the `download_buildpack` https://github.com/deis/deis/blob/master/builder/image/slugbuilder/builder/install-buildpacks#L6 function to make this working for branches, tags and specific commits.